app/models/package_manager/go.rb

```diff
@@ -243,34 +247,48 @@ def self.project_find_names(name)
 
       begin
         # https://go.dev/ref/mod#serving-from-proxy
-        go_import = get_html("https://#{name}?go-get=1", { request: { timeout: 2 } })
-          .xpath('//meta[@name="go-import"]')
-          .first
-          &.attribute("content")
-          &.value
-          &.split(" ")
-          &.last
-          &.sub(/https?:\/\//, "")
-
-        chosen_name = go_import&.start_with?(*KNOWN_HOSTS) ? [go_import] : [name]
+        # TODO: if nothing is being logged in GO_MODULES_PROJECT_FIND_NAMES_SUCCESS
+        # after a while, we might want to start using root_path instead of
+        # repo_url, if anything.
+        root_path = nil
+        vcs = nil
+        repo_url = nil
+        ms = Benchmark.ms do
+          root_path, vcs, repo_url = *get_html("https://#{name}?go-get=1", { request: { timeout: 2 } })
+            .xpath('//meta[@name="go-import"]')
+            .first
+            &.attribute("content")
+            &.value
+            &.split(" ")
+
+          repo_url = repo_url&.sub(/https?:\/\//, "")
+        end
+
+        chosen_name = repo_url&.start_with?(*KNOWN_HOSTS) ? [repo_url] : [name]
         # Collect info on what data we're mapping here, to ensure it's doing the right thing.
         StructuredLog.capture(
           "GO_MODULES_PROJECT_FIND_NAMES",
           {
             name: name,
-            found_name: go_import,
+            root_path: root_path,
+            vcs: vcs,
+            repo_url: repo_url,
+            found_name: repo_url,
             chosen_name: chosen_name,
+            go_lookup_duration: ms,
```

Comment:
would be helpful to aggregate how long we're spending in these "?go-get=1" requests overall